src/Effect.ts

```diff
@@ -4744,6 +4744,18 @@ export const zipWith: {
   ): Effect<R | R2, E | E2, B>
 } = fiberRuntime.zipWithOptions
 
+// -------------------------------------------------------------------------------------
+// applicatives
+// -------------------------------------------------------------------------------------
+/**
+ * @category combining
+ * @since 2.0.0
+ */
+export const ap: {
```

Comment:
the signature should be

```ts
export const ap: {
  <R2, E2, A>(that: Effect<R2, E2, A>): <R, E, B>(self: Effect<R, E, (a: A) => B>) => Effect<R | R2, E | E2, B>
  <R, E, A, B, R2, E2>(self: Effect<R, E, (a: A) => B>, that: Effect<R2, E2, A>): Effect<R | R2, E | E2, B>
} = dual(
  2,
  <R, E, A, B, R2, E2>(self: Effect<R, E, (a: A) => B>, that: Effect<R2, E2, A>): Effect<R | R2, E | E2, B> =>
    zipWith(self, that, (f, a) => f(a))
)
```

Reply:
Applied in https://github.com/Effect-TS/effect/pull/1504/commits/123a835195ad9476cc954b8a53a3099e672f928b, thanks @gcanti 🚀